server/server/src/lib.rs

```diff
@@ -216,18 +216,6 @@ async fn run_server(
         Ok(_) => {}
         Err(err) => {
             error!("Failed to perform the initial sync: {}", err);
-            if !is_develop() {
-                warn!("Falling back to bootstrap mode");
-                return run_stage0(
-                    settings,
-                    off_switch,
-                    token_bucket,
-                    token_secret,
-                    connection_manager,
-                    certificates,
-                )
-                .await;
-            }
```

Comment:
The original code makes sense to me. If we haven't synced yet we don't know if the settings are correct and we fall back to bootstrap mode be able to fix it. Because otherwise we might not be able to do the first login because we can't reach the central server and we are stuck.

In `RemoteDataSynchroniser::initial_pull` there is a check:
```rust
        if state.initial_remote_data_synced()? {
            return Ok(());
        }
```
However, this check isn't there in `Synchroniser::initial_pull`. This has been refactored recently, @andreievg just move the check into  `Synchroniser::initial_pull`? or was there some other intention behind it to always to the central pull?